frontend/src/pages/Worksheet.jsx

```diff
@@ -30,6 +30,35 @@ const StyledCalendarContainer = styled(SurfaceComponent)`
     color ${({ theme }) => theme.trans_dur};
 `;
 
+function WorksheetStats() {
+  const { courses, hidden_courses, cur_season } = useWorksheet();
+  let courseCnt = 0,
+    credits = 0,
+    workload = 0,
+    rating = 0;
+  for (const c of courses) {
+    if (hidden_courses[cur_season]?.[c.crn]) continue; // just skip hidden courses
+
+    courseCnt += 1;
+    credits += c.credits ?? 0;
+    workload += c.average_workload ?? 0;
+    rating += c.average_rating ?? 0;
+  }
+
+  const avgRating = courseCnt === 0 ? 0 : (rating / courseCnt).toFixed(2); // better to calculate not in the jsx
```

Comment:
There's absolutely no difference between this and inlining the expression in JSX, if you are speaking about efficiency. I can go with this for readability's sake, though, but let's remove the comment because there's no "one only way".

```suggestion
  const avgRating = courseCnt === 0 ? 0 : (rating / courseCnt).toFixed(2);
```

Reply:
🫡 🐐 👑